src/PublicApiAnalyzers/Core/Analyzers/DeclarePublicApiAnalyzer.Impl.cs

```diff
@@ -70,13 +99,15 @@ private sealed class Impl
 
             private readonly Compilation _compilation;
             private readonly ApiData _unshippedData;
+            private readonly bool _useNullability;
             private readonly ConcurrentDictionary<ITypeSymbol, bool> _typeCanBeExtendedCache = new ConcurrentDictionary<ITypeSymbol, bool>();
             private readonly ConcurrentDictionary<string, UnusedValue> _visitedApiList = new ConcurrentDictionary<string, UnusedValue>(StringComparer.Ordinal);
             private readonly IReadOnlyDictionary<string, ApiLine> _publicApiMap;
 
             internal Impl(Compilation compilation, ApiData shippedData, ApiData unshippedData)
             {
                 _compilation = compilation;
+                _useNullability = shippedData.NullableLines.Length > 0;
```

Comment:
Instead of enforcing that nullable enable is allowed only in shipped files, would it better to allow it in either of the shipped/unshipped files?